src/formatDate.js

```diff
@@ -8,7 +8,68 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  if (typeof date !== 'string' || date.length === 0) {
+    return '';
+  }
+
+  if (
+    !fromFormat ||
+    !toFormat ||
+    typeof fromFormat !== 'object' ||
+    typeof toFormat !== 'object' ||
+    !('length' in fromFormat) ||
+    !('length' in toFormat)
+  ) {
+    return '';
+  }
+
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  if (!fromSeparator || !toSeparator) {
+    return '';
+  }
+
+  const parts = date.split(fromSeparator);
+
+  if (parts.length !== 3) {
+    return '';
+  }
+
+  const partsMap = {
+    DD: '',
+    MM: '',
+    YYYY: '',
+    YY: '',
+  };
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    if (partsMap.hasOwnProperty(part)) {
+      partsMap[part] = parts[index];
+    }
+  });
+
+  if (partsMap['YY']) {
+    const yy = +partsMap['YY'];
+
+    if (yy < 30) {
+      partsMap['YYYY'] = '20' + partsMap['YY'];
+    } else {
+      partsMap['YYYY'] = '19' + partsMap['YY'];
+    }
+  }
```

Comment:
This block is responsible for converting a two-digit year to a four-digit year. However, this should only be done if the `toFormat` requires a `YYYY` format and the `fromFormat` was `YY`. You need to add a condition to check if the conversion is necessary based on the `toFormat` array.